### PR TITLE
Remove JITServer-specific methods for ROM strings access

### PR DESCRIPTION
**Pull request by dmitry-ten**
PR #11331 enabled JITServer to always cache ROM strings
and references.
This commit removes `TR_ResolvedJ9JITServerMethod::getROMString`
and `ClassInfo::getROMString` and removes overrides of methods
that used them.

Closes: #12721

### CODE REVIEW

**Comment by dmitry-ten** [comment]
@mpirvu ready for review.

File: runtime/compiler/env/j9method.cpp [hunk 1]
```diff
@@ -5666,7 +5666,7 @@ uint32_t
 TR_ResolvedJ9Method::classCPIndexOfMethod(uint32_t methodCPIndex)
    {
    uint32_t realCPIndex = jitGetRealCPIndex(_fe->vmThread(), romClassPtr(), methodCPIndex);
-   uint32_t classIndex = ((J9ROMMethodRef *) cp()->romConstantPool)[realCPIndex].classRefCPIndex;
```
Comment (mpirvu) [inline runtime/compiler/env/j9method.cpp:5669]: This change doesn't seem connected to the goal of this PR. Moreover, it's change for non-jitserver. Is it required?

**Review by mpirvu — COMMENTED** [review]
LGTH, but I favor leaving all those deleted methods from `TR_ResolvedJ9JITServerMethod` with a fatal assert inside.

File: runtime/compiler/env/j9method.cpp (hunk 1 above)
Comment (dmitry-ten) [reply runtime/compiler/env/j9method.cpp:5669]: This change allows us to remove JITServer version of `classCPIndexOfMethod`. It's related to ROM classes so I put it here, but it's not required to go into this PR.

**Comment by dmitry-ten** [comment]
> LGTH, but I favor leaving all those deleted methods from `TR_ResolvedJ9JITServerMethod` with a fatal assert inside.

I'm not sure what you mean by including a fatal assert. These methods must still be called on the server, it's just that their non-JITServer version now works correctly. If we override it for JITServer with asserts, then we'll just trigger asserts.

**Review by mpirvu — APPROVED** [review]
LGTM

**Comment by mpirvu** [comment]
jenkins test sanity plinuxjit,xlinuxjit,zlinuxjit jdk11

**Comment by mpirvu** [comment]
jenkins test sanity xlinux jdk11

**Pull request merged**